src/world/world_server.cpp

```diff
@@ -22,132 +22,18 @@ along with this program.  If not, see http://www.gnu.org/licenses/
-{
-    TracyZoneScoped;
-    TIMETYPE VanadielTOTD = CVanaTime::getInstance()->SyncTime();
-
-    // Weekly update for conquest (sunday at midnight)
-    static time_point lastConquestTally  = tick - 1h;
-    static time_point lastConquestUpdate = tick - 1h;
-    if (CVanaTime::getInstance()->getJstWeekDay() == 1 && CVanaTime::getInstance()->getJstHour() == 0 && CVanaTime::getInstance()->getJstMinute() == 0)
-    {
-        if (tick > (lastConquestTally + 1h))
-        {
-            conquest->updateWeekConquest();
-            lastConquestTally = tick;
-        }
-    }
-    // Hourly conquest update
-    else if (CVanaTime::getInstance()->getJstMinute() == 0)
-    {
-        if (tick > (lastConquestUpdate + 1h))
-        {
-            conquest->updateHourlyConquest();
-            lastConquestUpdate = tick;
-        }
-    }
-
-    // Vanadiel Hour
-    static time_point lastVHourlyUpdate = tick - 4800ms;
-    if (CVanaTime::getInstance()->getMinute() == 0)
-    {
-        if (tick > (lastVHourlyUpdate + 4800ms))
-        {
-            lastVHourlyUpdate = tick;
-        }
-    }
-
-    // JST Midnight
-    static time_point lastTickedJstMidnight = tick - 1h;
-    if (CVanaTime::getInstance()->getJstHour() == 0 && CVanaTime::getInstance()->getJstMinute() == 0)
-    {
-        if (tick > (lastTickedJstMidnight + 1h))
-        {
-            if (settings::get<bool>("main.ENABLE_DAILY_TALLY"))
-            {
-                UpdateDailyTallyPoints();
-            }
-
-            lastTickedJstMidnight = tick;
-        }
-    }
-
-    // 4-hour RoE Timed blocks
-    static time_point lastTickedRoeBlock = tick - 1h;
-    if (CVanaTime::getInstance()->getJstHour() % 4 == 0 && CVanaTime::getInstance()->getJstMinute() == 0)
-    {
-        if (tick > (lastTickedRoeBlock + 1h))
-        {
-            lastTickedRoeBlock = tick;
-        }
-    }
-
-    // Vanadiel Day
-    static time_point lastVDailyUpdate = tick - 4800ms;
-    if (CVanaTime::getInstance()->getHour() == 0 && CVanaTime::getInstance()->getMinute() == 0)
-    {
-        if (tick > (lastVDailyUpdate + 4800ms))
-        {
-            lastVDailyUpdate = tick;
-        }
-    }
-
-    if (VanadielTOTD != TIME_NONE)
-    {
-    }
-
-    return 0;
-}
+#include "time_server.h"
 
 WorldServer::WorldServer(int argc, char** argv)
 : Application("world", argc, argv)
-, conquestSystem(std::make_shared<ConquestSystem>())
+, sql(std::make_unique<SqlConnection>())
+, conquestSystem(std::make_unique<ConquestSystem>())
 , httpServer(std::make_unique<HTTPServer>())
 , messageServer(std::make_unique<message_server_wrapper_t>(std::ref(m_RequestExit)))
 {
-    // Anonymous namespace preparation
-    sql      = std::make_unique<SqlConnection>();
-    conquest = conquestSystem;
-
     // Tasks
-    CTaskMgr::getInstance()->AddTask("time_server", server_clock::now(), nullptr, CTaskMgr::TASK_INTERVAL, time_server, 2400ms);
+    std::shared_ptr<WorldServer> worldPtr(this);
```

Comment:
`world/main.cpp`
```cpp
int main(int argc, char** argv)
{
    auto pWorldServer = std::make_unique<WorldServer>(argc, argv);

    while (pWorldServer->IsRunning())
    {
        pWorldServer->Tick();
    }

    return 0;
}
```

WorldServer is already managed by a unique_ptr that's tied to the lifetime of the program. CTaskMgr is all static because it's set up as a singleton, so you don't need to worry about the lifetime of any of these object, so it's totally OK to just use `this` here.

If you WERE going to use shared_ptr from this here (if WorldServer was created by a shared_ptr in the first place), you'd need to use: https://en.cppreference.com/w/cpp/memory/enable_shared_from_this

Reply:
Done. Makes sense.